packages/pigeon/lib/swift_generator.dart

```diff
@@ -0,0 +1,642 @@
+            method.arguments.map((NamedType e) => e.name);
+        argSignature
+            .addAll(map2(argTypes, argNames, (String argType, String argName) {
+          return '$argName: $argType';
+        }));
+      }
+
+      final String returnType = method.returnType.isVoid
+          ? ''
+          : _nullsafeSwiftTypeForDartType(prefix, method.returnType);
+      if (method.isAsynchronous) {
+        argSignature.add('completion: @escaping ($returnType) -> Void');
+        indent.writeln('func ${method.name}(${argSignature.join(', ')})');
+      } else if (method.returnType.isVoid) {
+        indent.writeln('func ${method.name}(${argSignature.join(', ')})');
+      } else {
+        indent.writeln(
+            'func ${method.name}(${argSignature.join(', ')}) -> $returnType');
+      }
+    }
+  });
+
+  indent.addln('');
+  indent.writeln(
+      '/** Generated setup class from Pigeon to handle messages through the `binaryMessenger`.*/');
+  indent.write('class ${apiName}Setup ');
+  indent.scoped('{', '}', () {
+    final String codecName = _getCodecName(prefix, api);
+    indent.format('''
+/** The codec used by $apiName. */
+static var codec: FlutterStandardMessageCodec { $codecName.shared }
+''');
+    indent.writeln(
+        '/** Sets up an instance of `$apiName` to handle messages through the `binaryMessenger`. */');
+    indent.write(
+        'static func setup(binaryMessenger: FlutterBinaryMessenger, api: $apiName) ');
+    // aceitar api null
+    indent.scoped('{', '}', () {
+      for (final Method method in api.methods) {
+        final String channelName = makeChannelName(api, method);
+        final String varChannelName = '${method.name}Channel';
+
+        indent.writeln(
+            'let $varChannelName = FlutterBasicMessageChannel(name: "$channelName", binaryMessenger: binaryMessenger, codec: codec)');
+        indent.write('$varChannelName.setMessageHandler ');
+        indent.scoped('{ message, reply in', '}', () {
+          final List<String> methodArgument = <String>[];
+          if (method.arguments.isNotEmpty) {
+            indent.write('guard let args = message as? [Any?] ');
+            indent.scoped('else {', '}', () {
+              indent.writeln(
+                  'let error = FlutterError(code: "unexpected-args", message: "Unexpected argument parameters", details: nil)');
+              indent.writeln('reply(wrapError(error))');
+              indent.writeln('return');
+            });
+            indent.write('guard args.count == ${method.arguments.length} ');
+            indent.scoped('else {', '}', () {
+              indent.writeln(
+                  'let error = FlutterError(code: "unexpected-args-count", message: "Unexpected argument parameters count", details: "Expected parameters count: ${method.arguments.length}. Received: \\(args.count)")');
+              indent.writeln('reply(wrapError(error))');
+              indent.writeln('return');
+            });
```

Comment:
These to checks contribute to a ton of boilerplate in the generated code for a long list of methods. Per some recent discussion in `video_player` reviews, we should use helpers when possible. These two checks could be moved to a helper (taking the expected count as a parameter), cutting down on the repeated generated code.

Reply:
Although actually, why is the length a runtime check at all? It's not clear to me that we want to add a runtime cost of checks that can only fail if someone corrupts their generated code (e.g., by only updating one side).

Reply:
I got it. There is no reason to keep this runtime check. I remove it.

But this make me thinking about another check: the null safe check.
Is there need to this? Or can we presume that the type is correct either?

Follow some proposes to change in generator.
The force unwrap (`as!`) would be used just in case the type be non-null (`Type`).
To null type (`Type?`), would be used `as?`.

## Arguments parser
This
```swift
let valueArg = args[0] as! Type
```
instead of
```swift
guard let valueArg = args[0] as? Type else {
  let error = FlutterError(code: "unexpected-arg-type", message: "value argument unexpected type", details: "Expected type: Type. Received: \(type(of: args[0]))")
  reply(wrapError(error))
  return
}
```

## .fromMap
This
```swift
let value = map["value"] as! Type
```
instead of
```swift
guard let value = map["value"] as? Type else { return nil }
```

## CodeReader: FlutterStandardReader
This
```swift
return Type.fromMap(self.readValue() as! [String: Any?])  
```
instead of
```swift
guard let map = self.readValue() as? [String: Any?] else {
  return nil
}
return Type.fromMap(map)
```